_posts/2020-09-15-hive-acid-original-files.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: post
+title:  "Read support for original files of Hive transactional tables in Presto"
+author: Harmandeep Singh, Qubole
+excerpt_separator: <!--more-->
+---
+
+In [Presto 331](https://prestosql.io/docs/current/release/release-331.html),
+read support for Hive transactional tables was introduced. It works well, if a
+user creates a new Hive transactional table and reads it from Presto. However,
+if an existing table is converted to a Hive transactional table, Presto would
+fail to read data from such a table because read support for original files was
+missing. Original files are those files in a Hive transactional table that
+existed before the table was converted into a Hive transactional table.
+Until version 340, Presto expected all files in a Hive transactional table to be
+in Hive ACID format. Users would have to perform a major compaction to convert
+original files into ACID files (i.e. base files) in such tables. This is not
+always possible as the original flat table (table in non-ACID format) could be
+huge and converting all the existing data into ACID format can be very
+expensive.
+
+This blog is an extension of the blog [Hive ACID and transactional tables'
+support in Presto]({% post_url 2020-06-01-hive-acid %}). It first describes
+original files and then goes into details of read support for such files that
+was  added in Presto 340.
+
+<!--more-->
+
+# What are the original files?
+
+Files present in non-transactional ORC tables have the standard ORC schema. When
+a flat table is converted into a transactional table, existing files are not
+converted into Hive ACID format. Such files, in a transactional table, that are
+not in Hive ACID format, are called original files. These files are named as
+`000000_X`, `000000_X_copy_Y`. These files don’t have ACID columns and have
+differences in the schema as follows:
+
+Table Schema
```

Comment:
@mosabua : I have removed the table, as it was not getting rendered correctly because each cell contained complex struct. Now, it renders perfectly.

![image](https://user-images.githubusercontent.com/4463893/93223214-71357680-f78d-11ea-9aaa-e43f3d61eef1.png)